ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/AutomatonSimulator.java

```diff
@@ -436,8 +498,10 @@ Pair<PrePerimeterResult, Map<RangeAction<?>, Double>> shiftRangeActionsUntilFlow
         List<Pair<FlowCnec, Side>> flowCnecsWithNegativeMargin = getCnecsWithNegativeMarginWithoutExcludedCnecs(flowCnecs, flowCnecsToBeExcluded, automatonRangeActionOptimizationSensitivityAnalysisOutput);
 
         if (alignedRangeActions.stream().allMatch(HvdcRangeAction.class::isInstance) && !flowCnecsWithNegativeMargin.isEmpty()) {
-            // Disable AC emulation for HVDC lines, re-run sensitivity analysis and fetch new negative margins
-            automatonRangeActionOptimizationSensitivityAnalysisOutput = disableACEmulation(alignedRangeActions, network, preAutoPerimeterSensitivityAnalysis, automatonRangeActionOptimizationSensitivityAnalysisOutput, automatonState);
+            // Disable HvdcAngleDroopActivePowerControl for HVDC lines, fetch their set-point, re-run sensitivity analysis and fetch new negative margins
+            Pair<PrePerimeterResult, Map<HvdcRangeAction, Double>> result = disableHvdcAngleDroopActivePowerControl(alignedRangeActions, network, preAutoPerimeterSensitivityAnalysis, automatonRangeActionOptimizationSensitivityAnalysisOutput, automatonState);
+            automatonRangeActionOptimizationSensitivityAnalysisOutput = result.getLeft();
+            activatedRangeActionsWithSetpoint.putAll(result.getRight());
             flowCnecsWithNegativeMargin = getCnecsWithNegativeMarginWithoutExcludedCnecs(flowCnecs, flowCnecsToBeExcluded, automatonRangeActionOptimizationSensitivityAnalysisOutput);
```

Comment:
is there a risk that when disabling ac mode, we set a dc setpoint that is out of the range of the hvdc remedial action, all while making all the flow cnecs secure? (and therefore not entering the computeOptimalSetpoint that would put the setpoint back in its range)

Reply:
I added a watchdog in case it happens